### ISSUE TITLE
Proposal: remove --onefile from PyInstaller's build

### ISSUE THREAD

**Issue by eitchtee** [ChaoticAlexander/helldivers2-stratagem-activator#9]
Hello there @ChaoticAlexander

One thing is indispensable in the spread of managed democracy: Maximum Performance!

As such, I'd like to propose the removal of the `--onefile` argument when _compiling_ the main script.

The reason is quite simple: A single-file .exe created by PyInstaller is not really a single file, on the background  PyInstaller's runtime is extracting everything it needs to run that code on a temporary folder, running the code from there, and them removing the extracted files when done, which can impact startup time.

While this is perfectly fine on a normal application that is run once or twice a day, it can degrade quite quickly for a application that is essentially ran for 2 seconds or less multiple times in a short period, especially with a game and anti-cheat so CPU hungry as this one, from my point of view, any kind of unpacking happening on the background will be impacted, slowing startup time, especially on lesser hardware.

Now, I don't have any benchmark to back this up, but [PyInstaller's docs mentions the slower start time as a caveat of this argument.](https://www.pyinstaller.org/en/stable/operating-mode.html#bundling-to-one-file).

From my own anecdotal experience, on a gaming laptop that already struggles with the game in some moments, very often a stratagem will take _ages_ to register, which in turn makes me press the button again, and now I have two instances sending the same thing breaking everything, and I believe at least some of this slow startup is due to the behavior explained above.

---
In short, I understand the _sexiness_ of a single executable, but perhaps we could [bundle releases to one folder](https://www.pyinstaller.org/en/stable/operating-mode.html#bundling-to-one-folder) to get a shave a few milliseconds of startup.

**Comment by ChaoticAlexander** [comment]
@eitchtee I was actually thinking of this for some time now.
The performance difference is indeed quite noticeable.

I just uploaded the unpacked version as well; You can find it under releases :)

**Comment by ChaoticAlexander** [comment]
btw I used the --onedir for this, which I think provides a balance between performance and chaos haha
Let me know how it runs for you

**Comment by eitchtee** [comment]
Working perfectly. I think the onedir is a nice touch.

It could be placebo, but it feels a tad bit more responsive.

I recorded a small video for comparison, Supply Pack is onefile e Grenade Launcher is onedir, I added a ping sound when the button is pressed for comparison.

https://github.com/ChaoticAlexander/helldivers2-stratagem-activator/assets/10510126/32261a93-52eb-4720-8049-31740435724a

What do you think?

**Comment by eitchtee** [comment]
Those milliseconds add up I tell you haha

**Comment by ChaoticAlexander** [comment]
niceee, thanks for testing!
I do think it's quite an improvement, and it will probably solve some people's issues, now that I think of it.
and yeah, they do add up haha

btw keep in mind the delay being applied in-between key presses.
I initially added this as a failsafe for the script's detection.
It's working fine for me with 20-30ms between each press, but you might want to play around with that

Anyway, I'll close this issue as it seems the onedir did its trick hehe :)
